server/src/main/java/io/druid/segment/realtime/appenderator/Appenderator.java

```diff
@@ -61,8 +61,6 @@
    * This method may trigger a {@link #persistAll(Committer)} using the supplied Committer. If it does this, the
    * Committer is guaranteed to be *created* synchronously with the call to add, but will actually be used
    * asynchronously.
-   * <p>
-   * The add, clear, persist, persistAll, and push methods should all be called from the same thread.
```

Comment:
Hmm, I think these may still be needed. AppenderatorImpl's methods might not work properly if called concurrently from different threads. Also, nothing really needs to do that anyway. So I think we should leave in the restrictions on calling the major methods from the same thread.